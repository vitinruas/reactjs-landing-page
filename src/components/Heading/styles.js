import styled, { css } from 'styled-components'

export const Title = styled.h1`
  ${({ theme, appTheme }) =>
    css`
      color: ${appTheme === 'DARK' ? theme.colors.secondaryColor : theme.colors.primaryColor};
    `}
`
