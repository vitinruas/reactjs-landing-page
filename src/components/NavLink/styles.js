import styled, { css } from 'styled-components'

export const Container = styled.nav`
  ${({ theme, appTheme }) => css`
    background: ${appTheme === 'DARK' ? theme.colors.secondaryColor : theme.colors.primaryColor};
    display: flex;
    flex-flow: wrap row;

    @media ${theme.medias.lteMedium} {
      flex-flow: column nowrap;
      align-items: center;
    }
  `}
`
