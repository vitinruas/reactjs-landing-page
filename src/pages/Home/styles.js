import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  ${(props) => {
    return css`
      background: ${props.background};
    `
  }}
`
