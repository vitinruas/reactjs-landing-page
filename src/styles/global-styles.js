import { createGlobalStyle, css } from 'styled-components'

const GlobalStyles = createGlobalStyle`
  * {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
  }


  @media (min-width: 320px) and (max-width: 480px) {
    html {
      font-size: 75%; // 12
    }
  }

  @media (min-width: 481px) and (max-width: 768px) {
    html {
      font-size: 81.25%; // 13
    }
  }

  @media (min-width: 769px) and (max-width: 1024px) {
    html {
      font-size: 87.5%; // 14
    }
  }

  @media (min-width: 1025px) and (max-width: 1200px) {
    html {
      font-size: 93.75%; // 15
    }
  }

  @media (min-width: 1201) {
    html {
      font-size: 16px; // 16
    }
  }

  body {
    font-size: 1.6rem;
    font-family: ${({ theme }) =>
      css`
        ${theme.fonts.family.text}
      `}

  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${({ theme }) =>
      css`
        ${theme.fonts.family.title}
      `}
  }
`

export { GlobalStyles }
