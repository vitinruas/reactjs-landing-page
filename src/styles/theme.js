export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#FFFFFF',
    tertiaryColor: '#dc143c',

    lightBg: '#FFF',
    darkBg: '#0A1128',
  },
  fonts: {
    family: {
      text: "'Open Sans', sans-serif",
      title: "'Montserrat', sans-serif",
    },
    sizes: {
      small: '1.6rem',
      medium: '2.0rem',
      xmedium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      big: '4.8rem',
      xbig: '5.6rem',
      huge: '6.4rem',
      xhuge: '7.2rem',
    },
  },
  spacings: {
    small: '0.5rem',
    medium: '1rem',
    xmedium: '1.5rem',
    large: '2rem',
    xlarge: '2.5rem',
    big: '3.5rem',
    xbig: '4rem',
    huge: '4.5rem',
    xhuge: '5rem',
  },

  medias: {
    lteMedium: '(max-width: 768px)',
  },
}
