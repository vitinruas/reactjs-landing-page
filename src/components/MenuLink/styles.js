import styled, { css } from 'styled-components'
import { theme } from '../../styles/theme'

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacings.medium};
    color: ${theme.colors.primaryColor};
    position: relative;
    text-decoration: none;

    &:hover {
      cursor: pointer;
    }

    &::after {
      content: '';
      position: absolute;
      left: 50%;
      bottom: 0;
      transform: translateX(-50%);
      width: 0;
      height: 0.2rem;
      background-color: ${theme.colors.tertiaryColor};
      transition: all 0.5s;
    }
    &:hover::after {
      width: 50%;
    }
  `}
`
