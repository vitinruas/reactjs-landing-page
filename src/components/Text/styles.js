import styled, { css } from 'styled-components'
import { theme } from '../../styles/theme'
export const Container = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.medium};
  `}
`
