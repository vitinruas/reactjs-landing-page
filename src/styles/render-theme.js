import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { GlobalStyles } from './global-styles'
import { theme } from './theme'

export const renderTheme = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>,
  )
}
