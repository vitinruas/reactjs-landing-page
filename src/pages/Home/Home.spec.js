/* eslint-disable testing-library/no-debugging-utils */
/* eslint-disable testing-library/no-node-access */
import { screen } from '@testing-library/react'
import { Home } from '.'
import { renderTheme } from '../../styles/render-theme'

describe('<Home />', () => {
  test('Should have styles', () => {
    const { debug } = renderTheme(<Home />)

    const headingParent = screen.getByRole('heading', /hello, world!/i).parentElement

    // expect(heading).toHaveStyle({
    //   background: 'blue',
    //   fontSize: '2rem',
    // })

    expect(headingParent).toHaveStyleRule('background', 'blue')
    expect(headingParent).toHaveStyleRule('font-size', '2rem')
    expect(headingParent).toMatchSnapshot()
  })
})
