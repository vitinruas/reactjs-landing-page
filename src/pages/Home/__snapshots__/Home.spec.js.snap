// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Should have style 1`] = `
.c0 {
  background: blue;
  font-size: 2rem;
}

<section
  class="c0"
>
  <h1>
    Hello, World!
  </h1>
</section>
`;
